app/models/lookup_keys/puppetclass_lookup_key.rb

```diff
@@ -41,6 +41,10 @@ def puppet?
   def check_override_selected
     return if (changed - ['description', 'override']).empty?
     return if override?
+    return if !override? && override_changed? && (changed - ['merge_overrides',
```

Comment:
The first clause will always be true (since we check the opposite in the line before). Also, If the description also changed, this check will fail although it should not - please add a test for that case. 
I'm not too fond of very complex conditions as they are very error prone, I would be happy if you could find a way to simplify this check.